Salary App/Features/SalaryForm/View/SalaryFormViewController.swift

```diff
@@ -0,0 +1,336 @@
+        calculateBtn.translatesAutoresizingMaskIntoConstraints = false
+        calculateBtn.backgroundColor = .systemBlue
+        calculateBtn.setTitle("Submit", for: .normal)
+        calculateBtn.layer.cornerRadius = 16
+        calculateBtn.addTarget(self, action: #selector(calculateSalary), for: .touchUpInside)
+        return calculateBtn
+    }()
+    
+    
+   private let validationManager = ValidationManager()
+   private let taxCalculator = TaxCalculator()
+   private let niCalculator = NICalaculator()
+   private let studentLoanCalulator = StudentLoanCalculator()
+
+     let salaryvm = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
+   private let salaryTypeTextField = UITextField()
+ 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+    
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension SalaryFormViewController {
+    // MARK: Calculation
+    
+    @objc func calculateSalary() {
+        let salaryValue = Double(salaryTextField.text ?? "") ?? 0
+        let pensionValue = Double(pensionTextField.text ?? "") ?? 0
+         do {
+            try validationManager.isValid(texts: [salaryTextField.text ?? "" , pensionTextField.text ?? ""])
+            try validationManager.inValidNumber(values: [pensionTextField.text  ?? "" , salaryTextField.text ?? ""])
+            try validationManager.isValidSalary(number: salaryValue)
+            try validationManager.isValidPenContribution(number: pensionValue)
+             
+             let monthlySalary =  salaryvm.calculateMonthlySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+            
+             let dailySalary =  salaryvm.calculateDailySalary(salaryForm:SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+             
+             let weeklySalary = salaryvm.calculateWeeklySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+             
+             let hourlySalary = salaryvm.calculateHourlySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+
+          
+            let tax = taxCalculator.calculate(salary: salaryValue)
+            let ni = niCalculator.calculate(salary: salaryValue)
+            let studentLoan = studentLoanCalulator.calculate(salary: salaryValue)
+
+             let sheetViewController = SalaryBreakDownViewController(salaryBreakdown: SalaryBreakdown(monthlySalary: monthlySalary, dailySalary: dailySalary, hourlySalary: hourlySalary, weeklySalary: weeklySalary, tax: tax, ni: ni, studentLoan: studentLoan, originalSalary: salaryValue))
+            
+            navigationController?.pushViewController(sheetViewController, animated: true)
+            
+        } catch FormError.containsEmptyString {
```

Comment:
Rather than having multiple catch you can just have one and use a switch statement.

```

catch { error in

switch error {
  case error is FormError.containsEmptyString:
     showAlert(title: "Error", message: error.localizedDescription, actionTitle: "Ok")
....
}


}

```